project/app.cpp

```diff
@@ -0,0 +1,309 @@
+    }
+  }
+  if (data.setting_coord)
+    data.setting_coord =false;
+  int y_coord = 0;
+  for (int j = 0; j < stats.el.size(); ++j) {
+    if (stats.el.at(j).x == maxXCoord + GAPS)
+      y_coord = fmax(y_coord, stats.el.at(j).y + GAPS);
+  }
+  data.x = maxXCoord + 80;
+  data.y = y_coord;
+  data.set_coord =true;
+}
+
+void setInputsCoord(bench_statistics &stats, std::string name,
+                    int index_number) {
+  //    function that specifies coordinate for inputs
+  const int pos_x = 0;
+  const int pos_y =  (INPUT_HEIGHT+ GAPS)  * index_number;
+  for (int i = 0; i < stats.el.size(); ++i)
+    if (stats.el.at(i).name == name) {
+      stats.el.at(i).x = pos_x;
+      stats.el.at(i).y = pos_y;
+      stats.el.at(i).set_coord =true;
+    }
+}
+
+SDL_Window *sdlInitialization() {
+  //    initialization
+  if (SDL_Init(SDL_INIT_VIDEO) < 0) {
+    printf("SDL2 could not be initialized!\n"
+           "SDL2 Error: %s\n",
+           SDL_GetError());
+
+    return 0;
+  }
+  TTF_Init();
+
+#if defined linux && SDL_VERSION_ATLEAST(2, 0, 8)
+  // Disable compositor bypass
+  if (!SDL_SetHint(SDL_HINT_VIDEO_X11_NET_WM_BYPASS_COMPOSITOR, "0")) {
+    printf("SDL can not disable compositor bypass!\n");
+    return 0;
+  }
+#endif
+
+  // Create window
+  SDL_Window *window = SDL_CreateWindow(
+      "SDL2_ttf sample", SDL_WINDOWPOS_UNDEFINED, SDL_WINDOWPOS_UNDEFINED,
+      SCREEN_WIDTH, SCREEN_HEIGHT, SDL_WINDOW_SHOWN);
+  if (!window) {
+    printf("Window could not be created!\n"
+           "SDL_Error: %s\n",
+           SDL_GetError());
+  }
+  return window;
+}
+
+void eventLoop(bench_statistics &stats, SDL_Renderer *renderer,
+               std::vector<SDL_Rect> &elements) {
+  // Event loop exit flag
+  bool quit = false;
+
+  // Event loop
+  while (!quit) {
+    SDL_Event e;
+
+    // Wait indefinitely for the next available event
+    SDL_WaitEvent(&e);
+
+    // User requests quit
+    if (e.type == SDL_QUIT) {
+      quit = true;
+    }
+
+    // Initialize renderer color white for the background
+    SDL_SetRenderDrawColor(renderer, 0xFF, 0xFF, 0xFF, 0xFF);
+
+    // Clear screen
+    SDL_RenderClear(renderer);
+
+    // Set renderer color red to draw the square
+    SDL_SetRenderDrawColor(renderer, 0xFF, 0x00, 0x00, 0xFF);
+
+    // Draw filled square
+    for (int i = 0; i < stats.el.size(); ++i){
+        SDL_RenderFillRect(renderer,&elements[i]);
+    }
+    for (int i = 0; i < stats.el.size(); ++i) {
+        for (int j = 0; j < stats.el.at(i).gates.size(); ++j) {
+          SDL_SetRenderDrawColor(renderer, 0x00, 0x00, 0x00, 0xFF);
+          int el2 = stats.g[stats.el.at(i).gates.at(j)];
+          SDL_RenderDrawLine(renderer, stats.el.at(i).x, stats.el.at(i).y,
+                  stats.el.at(el2).x + INPUT_WIDTH,
+                  stats.el.at(el2).y + INPUT_HEIGHT);
+        }
+    }
+
+    // Update screen
+    SDL_RenderPresent(renderer);
+  }
+}
+
+void renderer(bench_statistics &stats, SDL_Window *window) {
```

Comment:
Bad naming: conventionally actions are named using verbs, e.g. `renderSomething`

Reply:
Renderer creation-destruction cycle and input data formatting are not related and must be dedicated procedures. Pass formatted data to the actual rendering procedure.